src/paas_charm/app.py

```diff
@@ -69,10 +72,31 @@ def should_run_scheduler(self) -> bool:
         return unit_id == "0"
 
 
+def generate_redis_env(relation_data: "PaaSRedisRelationData | None" = None) -> dict[str, str]:
```

Comment:
Would the type as `"PaaSRedisRelationData" | None` work?
I personally think that looks cleaner, since IDE would color the `| None` part correctly.
I am fine either way.